apps/ui/src/stores/notifications.ts

```diff
@@ -0,0 +1,127 @@
+import { defineStore } from 'pinia';
+import { getNetwork } from '@/networks';
+import { Network, ProposalsFilter } from '@/networks/types';
+import { NetworkID, Proposal } from '@/types';
+import pkg from '../../package.json';
+
+type Notification = {
+  id: string;
+  proposal: Proposal;
+  type: 'started' | 'ended';
+  timestamp: number;
+  unread: boolean;
+};
+
+const NOTIFICATION_TIME_WINDOW = 60 * 60 * 24 * 14; // 2 weeks
+const REFRESH_INTERVAL = 60 * 15; // 15 minutes
+let refreshNotificationInterval: NodeJS.Timeout;
+
+export const useNotificationsStore = defineStore('notifications', () => {
+  const loading = ref(true);
+  const notifications = ref<Notification[]>([]);
+  const lastUnreadTs = useStorage(
+    `${pkg.name}.notifications.last-unread`,
+    {} as Record<string, number>
+  );
+
+  const followedSpacesStore = useFollowedSpacesStore();
+  const metaStore = useMetaStore();
+  const { web3 } = useWeb3();
+
+  async function loadProposals(
+    network: Network,
+    current: number,
+    state: NonNullable<ProposalsFilter['state']>,
+    pivotTs: number,
+    spaceIds: string[]
+  ) {
+    return await network.api.loadProposals(spaceIds, { limit: 100 }, current, {
+      state,
+      start_gte: pivotTs
+    });
+  }
+
+  async function loadNotifications() {
+    const now = Math.floor(Date.now() / 1e3);
+    const pivotTs = now - NOTIFICATION_TIME_WINDOW;
+
+    if (!followedSpacesStore.followedSpacesIds.length) return;
+
+    (Object.keys(followedSpacesStore.followedSpaceIdsByNetwork) as NetworkID[]).forEach(
+      async networkId => {
+        await metaStore.fetchBlock(networkId);
+      }
+    );
+
+    const promises = (
+      Object.entries(followedSpacesStore.followedSpaceIdsByNetwork) as [NetworkID, string[]][]
+    )
+      .map(([networkId, spaceIds]) => {
+        const network = getNetwork(networkId);
+        const current = metaStore.getCurrent(networkId) ?? 0;
+
+        return [
+          loadProposals(network, current, 'active', pivotTs, spaceIds),
+          loadProposals(network, current, 'closed', pivotTs, spaceIds)
+        ];
```

Comment:
I think we can load those in single request per space instead of two, instead of using `active`/`closed` we can use this filter:
```gql
{
  start_gte: pivotTs, // only last two weeks
  start_lte: now, // only those that started already
}
```

This will half number of requests we make and also simplify this chunk of code (no more flattening in this step).

Reply:
Line 40 has been updated, there was an issue where closed proposals started more than 2 weeks ago were being filtered out.